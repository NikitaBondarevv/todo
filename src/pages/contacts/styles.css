.contacts {
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 39px 0 0;
  text-align: center;
  background: rgb(7 21 147 / 50%);
  width: 835px;
  color: #fff;
  font: 35px/40px "MyriadProRegular";
  margin: 0 auto 100px;
  letter-spacing: 1px;
  height: 487px;
}

.email {
  background: none;
  color: #fff;
  font-weight: bold;
  text-decoration: none;
  padding: 0 0 0 5px;
}

.email:hover {
  text-decoration: underline;
}
