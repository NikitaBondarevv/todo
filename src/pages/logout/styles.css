.logout {
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 39px 29px 104px;
  background: rgb(7 21 147 / 50%);
  width: 777px;
  height: 383px;
  color: #fff;
  font: 24px/40px "MyriadProRegular";
  margin: 0 auto 100px;
  letter-spacing: 1px;
}

.error {
  color: red;
  font-size: 25px;
  margin: 0 0 0 5px;
}
