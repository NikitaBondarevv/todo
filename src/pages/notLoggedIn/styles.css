.content {
  text-align: center;
  width: 775px;
  height: 358px;
  margin: 0 auto 100px;
  background: rgb(7 21 147 / 50%);
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 60px 30px 108px;
  font-size: 26px;
  line-height: 24px;
  color: #fff;
}

.content p {
  margin: 0;
  letter-spacing: 1px;
  margin: 0 0 160px;
}

.links {
  display: flex;
  justify-content: center;
  column-gap: 30px;
}

.link {
  display: block;
  color: #000;
  background: #fff;
  width: 244px;
  font-size: 22px;
  line-height: 55px;
  text-align: center;
  box-shadow: 0px 1px 7px #000;
  text-decoration: none;
}

.link:hover {
  text-decoration: underline;
}
