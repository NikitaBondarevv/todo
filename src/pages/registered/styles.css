.content {
  display: flex;
  flex-direction: column;
  width: 775px;
  height: 358px;
  margin: 0 auto 100px;
  background: rgb(7 21 147 / 50%);
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 60px 30px 108px;
  font-size: 26px;
  line-height: 24px;
  color: #fff;
}

.content p {
  margin: 0 0 40px;
  letter-spacing: 1px;
}

.link {
  display: block;
  color: #000;
  background: #fff;
  width: 244px;
  font-size: 22px;
  line-height: 55px;
  text-align: center;
  box-shadow: 0px 1px 7px #000;
  margin: 0 0 14px 59px;
  text-decoration: none;
  margin: 156px 0 0 50px;
}

.link:hover {
  text-decoration: underline;
}
