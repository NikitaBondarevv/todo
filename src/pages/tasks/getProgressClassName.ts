import styles from './styles.css'

export const getProgressClassName = (done: boolean | undefined) => {
  if (done) return styles.done

  if (done === undefined) return `${styles.progress} ${styles.updateTask}`

  return `${styles.waiting} ${styles.updateTask}`
}
