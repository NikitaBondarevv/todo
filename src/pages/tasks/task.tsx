import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { updateTask, deleteTask } from 'contracts/tasks'
import { getProgressClassName } from './getProgressClassName'
import { TTaskProps } from './types'
import styles from './styles.css'

export const Task = ({ data, getTasks }: TTaskProps) => {  
  const { done, title } = data

  const completeTask = async () => {
    await updateTask({ ...data, done: true })
    getTasks()
  }
  
  const setInProgress = async () => {
    await updateTask({ ...data, done: undefined })
    getTasks()
  }

  const removeTask = async () => {
    await deleteTask(data)
    getTasks()
  }

  return (
    <li className={getProgressClassName(done)}>
      &nbsp;
      <Link to={`/task/${data.day}/${data.id}`}>{title}</Link>
      {
        !done && (
          <div className={styles.buttons}>
            <button onClick={completeTask} type="button" />
            {
              done !== undefined && (
                  <button onClick={setInProgress} className={styles.inProgress} type="button" />
                )
            }
            <button onClick={removeTask} className={styles.delete} type="button" />
          </div>
        )
      }
    </li>
  )
}

Task.defaultProps = {
  data: {},
}

Task.propTypes = {
  data: PropTypes.object,
  getTasks: PropTypes.func
}
