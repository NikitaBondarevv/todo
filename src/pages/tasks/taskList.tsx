import { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { Task } from './task'
import { TTaskListProps } from './types'
import styles from './styles.css'

export const TaskList = ({ tasks, getTasks, activeTabIndex }: TTaskListProps) => {
  const [todos, setTodos] = useState(tasks)

  useEffect(() => {
    setTodos(tasks)
  }, [tasks])

  return (
    <div>
      <ol className={styles.tasksList}>
        {
          todos.map((task, index) => <Task key={index} data={task} getTasks={getTasks} />)
        }
      </ol>
      <Link to={`/task/${activeTabIndex}`} className={styles.addTask}>Add new task</Link>
    </div>
  )
}

TaskList.defaultProps = {
  tasks: [],
  activeTabIndex: 0,
  getTasks: () => {}
}

TaskList.propTypes = {
  tasks: PropTypes.array,
  getTasks: PropTypes.func,
  activeTabIndex: PropTypes.number
}
