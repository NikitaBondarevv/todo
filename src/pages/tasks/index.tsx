import { useState, useEffect } from 'react'

import { daysOfTheWeek } from 'helpers/constans'
import { getTasks } from 'contracts/getTasks'
import { Preloader } from 'components/preloader'
import { TaskList } from './taskList'
import { Tabs } from 'components/tabs'
import { Tab } from 'components/tabs/tab'

export const Tasks = () => {
  const [days, setDays] = useState([])
  const [isLoading, setIsloading] = useState(false)

  const getDailyTasks = async () => {
    setDays(await getTasks())
    setIsloading(false)
  }

  useEffect(() => {
    setIsloading(true)
    getDailyTasks()
  }, [])

  const getCurrentDay = () => {
    const dayNow = new Date().getDay()

    return dayNow ? dayNow - 1 : 6
  }

  return (
    isLoading ? <Preloader size={100} />
      : <Tabs selectedIndex={getCurrentDay()}>
        {
          days.map((day, index) => (
            <Tab key={index} title={daysOfTheWeek[index]}>
              <TaskList tasks={day} getTasks={getDailyTasks} activeTabIndex={index} />
            </Tab>
          ))
        }
      </Tabs>
  )
}
