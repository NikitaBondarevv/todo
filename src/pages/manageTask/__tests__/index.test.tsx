import { fireEvent, screen, act } from '@testing-library/react'
import { createTask, updateTask } from 'contracts/tasks'

import { customRender } from '__mocks__/customRender'
import { ManageTask } from '..'
import styles from '../styles.css'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    day: '1',
    id: '1'
  })
}))

jest.mock('contracts/tasks', () => ({
  createTask: jest.fn(),
  getTaskById: () => ({
    title: 'test',
    description: 'test',
    id: '1',
    day: '1'
  }),
  updateTask: jest.fn()
}))

describe('<ManageTask />', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const testText = 'test'

  test('should match snapshot', () => {
    const { asFragment } = customRender(<ManageTask />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should fire onChange() on input and should be set to text "test"', async () => {
    const { container } = customRender(<ManageTask />)
    const input = container.querySelector(`.${styles.title}`)

   await act(() => fireEvent.change(input!, { target: { value: testText } }))
    expect(screen.getByDisplayValue(testText)).not.toBe('')
  })

  test('should fire onChange() on textarea and should be set to text "test"', async () => {
    const { container } = customRender(<ManageTask />)
    const textarea = container.querySelector(`.${styles.description}`)

    await act(() => fireEvent.change(textarea!, { target: { value: testText } }))

    expect(screen.getByDisplayValue(testText)).not.toBe('')
  })

  test('should not fire createTask() in the onSubmit event if the length of the value is less than 2', async () => {
    const { container } = customRender(<ManageTask />)
    const form = container.querySelector(`.${styles.updateCreateTask}`)
    const input = container.querySelector(`.${styles.title}`)

    fireEvent.change(input!, { target: { value: 'a' } })
    await act(() => fireEvent.submit(form!))

    expect(createTask).not.toHaveBeenCalled()
  })

  test('should fire createTask() if id is not present', async () => {
    const { container } = customRender(<ManageTask />)
    const form = container.querySelector(`.${styles.updateCreateTask}`)

    await act(() => fireEvent.submit(form!))

    expect(createTask).toHaveBeenCalled()
  })

  test('should fire createTask() if id is present', async () => {
    const { container } = customRender(<ManageTask />)
    const form = container.querySelector(`.${styles.updateCreateTask}`)

    await act(() => fireEvent.submit(form!))

    expect(updateTask).toHaveBeenCalled()
  })
})
