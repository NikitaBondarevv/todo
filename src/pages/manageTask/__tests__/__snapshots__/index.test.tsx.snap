// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageTask /> should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="updateCreateTask"
  >
    <input
      class="title"
      name="text"
      value="New task"
    />
    <textarea
      class="description"
      name="description"
      placeholder="Add description here"
    />
    <input
      type="submit"
      value="SAVE"
    />
  </form>
</DocumentFragment>
`;
